src/types/types.ts

```diff
@@ -1,3 +1,171 @@
+export type TimeInterval = {
+    duration: string
+    end: string
+    start: string
+}
+
+export type CustomFieldValues = {
+    customFieldId: string
+    name: string
+    timeEntryId: string
+    type: string
+    value: string
+}
+
+export type TimeEntries = {
+    customFieldValues: CustomFieldValues[]
+    description: string
+    id: string
+    isLocked: boolean
+    kioskId: string | null
+    projectId: string | null
+    taskId: string | null
+    timeInterval: TimeInterval
+    type: string
+    userId: string
+    workspaceId: string
+}
+
+export type Page = {
+    pageSize: number
+    page: number
+}
+
+export type TimeEntryDataProp = {
+    description: string
+    start: string
+    end: string
+    projectId: string | null
+}
+
+export type IdProp = {
+    id: string
+}
+
+export type UpdateTimeEntryProp = TimeEntryDataProp & IdProp
+
+export type SelectedOption = {
+    value: string
+    label: string
+}
+
+export type CurrentTask = {
+    startTime: string
+    endTime: string
+    duration: string
+    taskName: string
+    project: string
+    client: string
+}
+
+export type Project = {
+    clientId: string
+    clientName: string
+    color: string
+    duration: string
+    id: string
+    name: string
+}
+
+export type ProjectId = {
+    projectId: string | null
```

Comment:
basically this is an `optional` value is it? If yes are there other ways to do this instead of `string | null`?

Reply:
https://github.com/PreethiKalyani99/Clockify-Typescript/commit/3964a5e5d151c1eeff87bc3d31529cecd8f2a0b5

changed ProjectId as optional value.

Reply:
How is `projectId?: string` is different from this?

Reply:
If I use projectId?: string, I would have to explicitly check whether projectId is present or not. That's why I define the projectId type as string | null, so I can pass null to the API when there is no projectId, Anna

Reply:
you mean you are getting error from API when the `projectId` is `undefined`?

Reply:
yes Anna